hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/SortPhysicalRule.java

```diff
@@ -64,19 +63,7 @@ public void onMatch(RelOptRuleCall call) {
                 sortTransforms.add(convert(logicalSort));
             } else {
                 if (logicalSort.offset != null || logicalSort.fetch != null) {
-                    SortPhysicalRel top = (SortPhysicalRel) convert(logicalSort);
-                    // TODO: [sasha], [Hakan] additional Sort relation is used for FETCH OFFSET with index scan.
-                    // TODO: It's required for correct work of index scan, but should be optimized in next revisions.
-                    top = new SortPhysicalRel(
-                            top.getCluster(),
-                            top.getTraitSet(),
-                            physicalInput,
-                            top.getCollation(),
-                            top.offset,
-                            top.fetch,
-                            top.getRowType()
-                    );
-                    nonSortTransforms.add(top);
+                    nonSortTransforms.add(convert(logicalSort));
```

Comment:
We can create GitHub issue to highlight required optimisation of FETCH OFFSET special case.